### PR TITLE
Allow linchpin destroy to remove libvirt networks

### PR DESCRIPTION
**Pull request by 14rcole**
Adds a key to the libvirt network schema, delete_on_destroy, that destroys and undefines libvirt networks if true

### CODE REVIEW

**Comment by 14rcole** [comment]
@mcornea fixes #940

File: docs/source/libvirt.rst [hunk 1]
```diff
@@ -20,48 +20,51 @@ Within Linchpin, the :term:`libvirt_node` :term:`resource_definition` has more
...
+| cloud_config       | false | list     | role          | http://cloudinit.readthedocs.io                        |
+|                    |       |          |               | is used here                                           |
++--------------------+-------+----------+---------------+---------------------+----------------------------------+
+| additional_storage | false | string   | role          | 1G                  |                                  |
++--------------------+-------+----------+---------------+---------------------+----------------------------------+
+| uri                | false | string   | module: uri   | qemu:///system      |                                  |
++--------------------+-------+----------+---------------+---------------------+----------------------------------+
+| count              | false | string   | N/A           |                     |                                  |
++--------------------+-------+----------+---------------+---------------------+----------------------------------+
+| delete_on_destroy  | false | boolean  | N/A           | False               | If true, libvirt destroy will    |
```
Comment (samvarankashyap) [inline docs/source/libvirt.rst]: @14rcole I think delete_on_destroy should also be on libvirt_network documentation, shouldn't it?

**Comment by mcornea** [comment]
@14rcole I tested the change and it worked fine for me. Thanks!

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**